mini-projects/01-device-driver/pico-image/src/main.rs

```diff
@@ -246,25 +250,53 @@ fn main() -> ! {
                 if current_time - last_button_state_transmission_time > SERIAL_TX_PERIOD {
                     last_button_state_transmission_time = current_time;
                     let mut button_text: String<20> = String::new();
-                    let button_data = (left_button
+                    let button_data = (ne_button
                         .is_high()
                         .expect("GPIOs should never fail to read state")
                         as u8)
-                        + ((top_button
+                        + ((ne_button
                             .is_high()
                             .expect("GPIOs should never fail to read state")
                             as u8)
                             << 1)
-                        + ((bottom_button
+                        + ((se_button
                             .is_high()
                             .expect("GPIOs should never fail to read state")
                             as u8)
                             << 2)
-                        + ((right_button
+                        + ((sw_button
+                            .is_high()
+                            .expect("GPIOs should never fail to read state")
+                            as u8)
+                            << 3)
+                        + ((nw_button
                             .is_high()
                             .expect("GPIOs should never fail to read state")
                             as u8)
-                            << 3);
+                            << 4)
+                        + ((n_button
+                            .is_high()
+                            .expect("GPIOs should never fail to read state")
+                            as u8)
+                            << 5)
+                        + ((s_button
+                            .is_high()
+                            .expect("GPIOs should never fail to read state")
+                            as u8)
+                            << 6)
+                        + ((e_button
+                            .is_high()
+                            .expect("GPIOs should never fail to read state")
+                            as u8)
+                            << 7)                            
+                        + ((w_button
+                            .is_high()
+                            .expect("GPIOs should never fail to read state")
+                            as u8)
+                            << 8)                        
```

Comment:
this doesn't compile since it could overflow - I assume this was just something that the fix didn't get pushed?